File: garden/seeds/java/app/src/main/java/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/module/AppServicesComponent.java
package com.{{company_name}}.android.{{app_package_name_prefix}}.module;

import com.{{company_name}}.android.{{app_package_name_prefix}}.mvp.presenter.HomePresenter;
import javax.inject.Singleton;

import dagger.Component;

/**
 * Dagger component to provide dependency injection
 */
@Singleton
@Component(modules = {{app_class_prefix}}Module.class)
public interface AppServicesComponent {

	void inject(HomePresenter presenter);
}

File: garden/seeds/kotlin/app/src/main/kotlin/com/WATER_ME_WITH_COMPANY/android/WATER_ME_WITH_APP_NAME/module/AppServicesComponent.kt
package com.{{company_name}}.android.{{app_package_name_prefix}}.module

import com.{{company_name}}.android.{{app_package_name_prefix}}.mvp.presenter.MainPresenter
import dagger.Component
import javax.inject.Singleton

/**
 * Dagger component to provide dependency injection
 */
Singleton
Component(modules = arrayOf({{app_class_prefix}}Module::class))
public interface AppServicesComponent {

    fun inject(presenter: MainPresenter)

}